lms/djangoapps/certificates/views/webview.py

```diff
@@ -588,46 +546,155 @@ def render_html_view(request, user_id, course_id):
+        # Append site configuration overrides
+        _update_configuration_context(context, configuration)
 
-    # Append/Override the existing view context values with any course-specific static values from Advanced Settings
-    context.update(course.cert_html_view_overrides)
+        # Add certificate header/footer data to current context
+        context.update(get_certificate_header_context(is_secure=request.is_secure()))
+        context.update(get_certificate_footer_context())
 
-    # Track certificate view events
-    _track_certificate_events(request, context, course, user, user_certificate)
+        # Append/Override the existing view context values with any course-specific static values from Advanced Settings
+        context.update(course.cert_html_view_overrides)
 
-    # FINALLY, render appropriate certificate
-    return _render_certificate_template(request, context, course, user_certificate)
+        # Track certificate view events
+        _track_certificate_events(request, context, course, user, user_certificate)
+
+        # Render the certificate
+        return _render_valid_certificate(request, context, custom_template)
+
+
+def _get_catalog_data_for_course(course_key):
+    """
+    Retrieve data from the Discovery service necessary for rendering a certificate for a specific course.
+    """
+    course_certificate_settings = CertificateGenerationCourseSetting.get(course_key)
+    if not course_certificate_settings:
+        return {}
+
+    catalog_data = {}
+    course_run_fields = []
+    if course_certificate_settings.language_specific_templates_enabled:
+        course_run_fields.append('content_language')
+    if course_certificate_settings.include_hours_of_effort:
+        course_run_fields.extend(['weeks_to_complete', 'max_effort'])
+
+    if course_run_fields:
+        course_run_data = get_course_run_details(course_key, course_run_fields)
+        if course_run_data.get('weeks_to_complete') and course_run_data.get('max_effort'):
+            try:
+                weeks_to_complete = int(course_run_data['weeks_to_complete'])
+                max_effort = int(course_run_data['max_effort'])
+                catalog_data['hours_of_effort'] = weeks_to_complete * max_effort
+            except ValueError:
+                log.exception('Error occurred while parsing course run details')
+        catalog_data['content_language'] = course_run_data.get('content_language')
+
+    return catalog_data
+
+
+def _get_custom_template_and_language(course_id, course_mode, course_language):
+    """
+    Return the custom certificate template, if any, that should be rendered for the provided course/mode/language
+    combination, along with the language that should be used to render that template.
+    """
+    closest_released_language = _get_closest_released_language(course_language) if course_language else None
+    template = get_certificate_template(course_id, course_mode, closest_released_language)
```

Comment:
Note: This change means that custom templates in languages that are not officially supported by the LMS (ie languages that haven't been added to DarkLang) will not be used. @bderusha do we need to communicate this in some way to stakeholders?